src/relayer/Relayer.ts

```
@@ -35,10 +35,19 @@ export class Relayer {
     // Fetch all unfilled deposits, order by total earnable fee.
     // TODO: Note this does not consider the price of the token which will be added once the profitability module is
     // added to this bot.
-    const unfilledDeposits = getUnfilledDeposits(this.clients.spokePoolClients, this.maxUnfilledDepositLookBack).sort(
-      (a, b) =>
+
+    // Require that all fillable deposits meet the minimum specified number of confirmations.
+    const unfilledDeposits = getUnfilledDeposits(this.clients.spokePoolClients, this.maxUnfilledDepositLookBack)
+      .filter((x) => {
+        return (
+          x.deposit.originBlockNumber <=
+          this.clients.spokePoolClients[x.deposit.originChainId].latestBlockNumber -
+            (this.config.minDepositConfirmations[x.deposit.originChainId] ?? 0)
```

Just to confirm, is it correct that the fallback to 0 is only required to make the tests work?

---

well if the `minDepositConfirmations` for a chain isn't set it might fail since subtracting `undefined` could be an issue?

---

I _think_ we should be protected against an undefined minDepositConfirmations in normal scenarios, because we assert() on their existence when reading in the configuration: https://github.com/across-protocol/relayer-v2/pull/302/files#diff-cfd7a6f4fb96bcb633a6a603efa2cc1ac9fe6879bfafdef1c3a935be188e27db

The justification for this is that a user might define an alternative configuration for themselves, and then we add a new chain, and their environment suddenly silently doesn't handle that. I tried to remove this fallback to 0 previously, but found that it created issues in Dataworker tests (...and I remember being confused as to why this should impact the Dataworker!).

I think it'd be nice to remove the implicit fallback here (or at least revert to a default value from Constants). It's not necessarily a part of this change, though :)

---

We could also verify this when ingesting the env variable rather than doing it here.

---

I think the fallback to 0 is useful though in this exact situation, right?

> a user might define an alternative configuration for themselves, and then we add a new chain, and their environment suddenly silently doesn't handle that.

---

> I think the fallback to 0 is useful though in this exact situation, right?

My preference is to just fail hard and fast if we detect this situation, and that's how it currently works. If we decided to allow for a silent fallback, I'd opt for falling back to our own internal default for the specific chain, rather than 0.

So, this is a bit of an academic discussion, since we _should_ never encounter this behaviour in practice given the current strictness of the config validation. I just wanted to take the opportunity to discuss potentially removing/adjusting this fallback as well, which would imply that we need to introduce minDepositConfirmations support in our tests. I looked at this briefly but was seeing test chain IDs and never found where they were defined.

---

I've removed the fallback and fixed tests :)